### DISCUSSION TITLE
[Request] Jinja Support

### DISCUSSION THREAD

**Discussion by Diegovsky** [wrabit/django-cotton#220] [Ideas]
While DTL is good enough, we use jinja a lot internally. Could django-cotton support jinja someday?

**Comment by wrabit** [comment]
@Diegovsky It probably could if enough people requested it. (You’re the first!). Besides that, as it’s something I don’t use - it’s not planned at this stage. I will move this to a discussion to see if interest gathers over time.

**Comment by wrabit** [comment]
+1 from https://en.tgchannels.org/channel/reorx_share?size=30&lang=all&first=5205&start=5205

> Although I like to write pure backend rendering websites, I have to admit that JSX is much more intuitive than Django and Jinja2 templates. The latter always have countless syntax sugars, which seem powerful, but can easily become a mess. As the requirements become more and more complex, the code often becomes difficult to maintain, making people want to refactor it into SPA or Node SSR applications.
> 
> django-cotton introduces front-end component templates into Django, bringing new ideas to backend rendering projects. I hope Jinja2 can also have a similar approach, but the best thing is to have a Python template engine implemented from scratch based on this idea.
> 
> https://django-cotton.com/

**Comment by wrabit** [comment]
+1 https://news.ycombinator.com/item?id=42006842

**Comment by Diegovsky** [comment]
Please show support by upvoting my toplevel post, thanks :)

**Comment by davisums** [comment]
Also very interested in this!

**Comment by jabir-git** [comment]
me too!

**Comment by wrabit** [comment]
I just discovered this https://www.reddit.com/r/flask/comments/1kujybr/jinja_ui_components/
 
Which led me to a [basecoatui](https://basecoatui.com/) which has shadcn/ui implemented with alpine.js + tailwind.

And has [some macro's](https://basecoatui.com/installation/#install-macros) you can use in jinja.

Just curious if it's enough for those who desire this?

**Reply by jabir-git** [reply]
yes it is abviously very interesting, thanks!

**Comment by wrabit** [comment]
Demo code for a button in cotton for jinja could look something like this:

<img width="807" height="461" alt="image" src="https://github.com/user-attachments/assets/8989a571-6d01-48d3-ac05-6c30192d07a5" />

How does it look to those who are using jinja? 

Django cotton would do something like this instead with the c-vars since DTL doesn't have the set helper:

<img width="835" height="478" alt="image" src="https://github.com/user-attachments/assets/f0b3c610-f710-4020-9d18-496c2071521a" />

Is there any benefit in supporting the DTL version of c-vars?
